src/FilesystemRepository.php

```diff
@@ -349,6 +378,18 @@ private function removeResource($filesystemPath, &$removed)
         $this->filesystem->remove($filesystemPath);
     }
 
+    private function moveResource($sourcePath, $targetPath, &$moved)
+    {
+        $targetPathInBaseDir = $this->baseDir.$targetPath;
+        ++$moved;
+
+        if (is_dir($sourcePath)) {
+            $moved += $this->countChildren($sourcePath);
+        }
+
+        $this->filesystem->rename($sourcePath, $targetPathInBaseDir);
```

Comment:
as the filesystem uses `rename()`, I believe there is no rollback feature needed here